### PR TITLE
Make iSCSI and Fibre Channel grains optional

### PR DESCRIPTION
**Pull request by twangboy**
### What does this PR do?
Creates `iscsi.py` and `fibre_channel.py`
Removes those grains from `core.py`
Gates the grains with options from the config file:
- `fibre_channel_grains`
- `iscsi_grains`

### What issues does this PR fix or reference?
https://github.com/saltstack/salt/issues/46425
https://github.com/saltstack/salt/pull/43647

### Previous Behavior
Grains would (attempt) to be populated every time. It is especially slow on Windows because it uses PowerShell and WMIC.

### New Behavior
These grains are now optional via config

### Tests written?
No

### Commits signed with GPG?
Yes

### CODE REVIEW

**Comment by rallytime** [comment]
@twangboy This is causing the following test to fail:

- unit.grains.test_iscsi.IscsiGrainsTestCase.test_windows_iscsi_iqn_grains

https://jenkins.saltstack.com/job/PR/job/salt-pr-linode-ubuntu14-n/20844/#showFailuresLink

**Comment by twangboy** [comment]
@rallytime I think this is fixed now.

File: salt/grains/iscsi.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+import logging
+import os
+
+# Import Salt libs
+import salt.modules.cmdmod
+import salt.utils.files
+import salt.utils.path
+import salt.utils.platform
+
+__proxyenabled__ = ['iscsi']
```
Comment (rallytime) [inline salt/grains/iscsi.py]: Are these grains for proxy minions?

File: salt/grains/iscsi.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+__virtualname__ = 'iscsi'
+
+# Get logging started
+log = logging.getLogger(__name__)
+
+
+def __virtual__():
+    if __opts__.get('iscsi_grains', False) is False:
```
Comment (rallytime) [inline salt/grains/iscsi.py]: Is this setting documented anywhere?

File: salt/grains/fibre_channel.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+import glob
+import logging
+
+# Import Salt libs
+import salt.modules.cmdmod
+import salt.utils.platform
+import salt.utils.files
+
+__proxyenabled__ = ['fibre_channel']
```
Comment (rallytime) [inline salt/grains/fibre_channel.py]: Are these grains for proxy minions?

File: salt/grains/fibre_channel.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+__virtualname__ = 'fibre_channel'
+
+# Get logging started
+log = logging.getLogger(__name__)
+
+
+def __virtual__():
+    if __opts__.get('fibre_channel_grains', False) is False:
```
Comment (rallytime) [inline salt/grains/fibre_channel.py]: Is this setting documented anywhere? I know it's documented in this file, but i mean in other places like config options, release notes, etc.

**Review by rallytime — CHANGES_REQUESTED** [review]
I have a couple of small questions/comments. Otherwise, looks good.

File: salt/grains/iscsi.py (hunk 1 above)
Comment (twangboy) [reply salt/grains/iscsi.py]: I don't know, I was just copying other tests. I thought that was for alternate names or something.

File: salt/grains/iscsi.py (hunk 2 above)
Comment (twangboy) [reply salt/grains/iscsi.py]: No

File: salt/grains/fibre_channel.py (hunk 3 above)
Comment (twangboy) [reply salt/grains/fibre_channel.py]: See above

File: salt/grains/fibre_channel.py (hunk 4 above)
Comment (twangboy) [reply salt/grains/fibre_channel.py]: No

File: salt/grains/iscsi.py (hunk 2 above)
Comment (twangboy) [reply salt/grains/iscsi.py]: It is now

File: salt/grains/fibre_channel.py (hunk 4 above)
Comment (twangboy) [reply salt/grains/fibre_channel.py]: Also documented

**Review by rallytime — APPROVED** [review]

**Comment by rallytime** [comment]
re-run py

**Pull request merged**